pkg/engine/templates_generated.go

```diff
@@ -18854,24 +18855,42 @@ func k8sWindowsconfigfuncPs1() (*asset, error) {
 
 var _k8sWindowscontainerdfuncPs1 = []byte(`# this is $global to persist across all functions since this is dot-sourced
 $global:ContainerdInstallLocation = "$Env:ProgramFiles\containerd"
+$global:Containerdbinary = (Join-Path $global:ContainerdInstallLocation containerd.exe)
 
 function RegisterContainerDService {
-  Assert-FileExists (Join-Path $global:ContainerdInstallLocation containerd.exe)
+  Param(
+    [Parameter(Mandatory = $true)][string]
+    $kubedir
+  )
 
-  Write-Host "Registering containerd as a service"
-  $cdbinary = Join-Path $global:ContainerdInstallLocation containerd.exe
-  $svc = Get-Service -Name containerd -ErrorAction SilentlyContinue
+  Assert-FileExists $global:Containerdbinary
+
+  # in the past service was not installed via nssm so remove it in case
+  $svc = Get-Service -Name "containerd" -ErrorAction SilentlyContinue
   if ($null -ne $svc) {
-    & $cdbinary --unregister-service
+    sc.exe delete containerd
   }
-  & $cdbinary --register-service
+
+  Write-Host "Registering containerd as a service"
+  # setup containerd
+  & "$KubeDir\nssm.exe" install containerd $global:Containerdbinary | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd AppDirectory $KubeDir | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd DisplayName containerd | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd Description containerd | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd Start SERVICE_DEMAND_START | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd ObjectName LocalSystem | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd Type SERVICE_WIN32_OWN_PROCESS | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd AppThrottle 1500 | RemoveNulls
+  & "$KubeDir\nssm.exe" set containerd AppStdout C:\k\containerd.log | RemoveNulls
```

Comment:
Should we log to the same directory as containerd or to c:\k\?
If we log to c:\k\ logs will all be together but if we log next to containerd (c:\program files\contaienrd) it might be less confusing for someone looking for the logs...

Reply:
Also, should we use $KubeDir instead of c:\k\?

Reply:
I was leaning towards `c:\k` but don't have strong feelings.  I think the hardcoded `c:\k` was copy past from testing. I will update that.

Reply:
I set the working dir to c:\k as well to fix the azure cni logging issue